docs/contributing/contributing-code/contributing-code-debugging/radius-os-processes-debugging.md

```diff
@@ -0,0 +1,405 @@
+# Running Radius as OS Processes for Advanced Debugging
+
+This guide details how to leverage the fully-integrated VS Code debugging experience for Radius development. By running core components as native OS processes instead of in containers, you can take advantage of pre-configured launch configurations and tasks to enable a seamless "inner-loop" workflow. This setup allows for advanced debugging capabilities, such as setting breakpoints, inspecting variables, and stepping through code in real-time—all directly within the VS Code editor—significantly accelerating development and troubleshooting.
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Quick Start](#quick-start)
+3. [Prerequisites](#prerequisites)
+4. [Development Workflow](#development-workflow)
+5. [VS Code Debugging](#vs-code-debugging)
+6. [Troubleshooting](#troubleshooting)
+
+## Overview
+
+Radius consists of several key components that normally run in Kubernetes containers:
+- **Applications Resource Provider (applications-rp)** - Manages Applications.Core resources
+- **UCP Daemon (ucpd)** - Universal Control Plane for resource management
+- **Controller** - Kubernetes controller for managing Radius resources
+- **Dynamic Resource Provider (dynamic-rp)** - Handles dynamic resource types
+
+Running these as OS processes enables:
+- Full debugger support with breakpoints and variable inspection
+- Real-time configuration changes
+- Performance profiling and analysis
+- Network traffic inspection
+
+## Quick Start
+
+The simplest way to get debugging working:
+
+### Prerequisites
+- Local Kubernetes cluster accessible via kubectl (Docker Desktop, k3d, minikube, etc.)
+- Current kubectl context must point to your **local** cluster
+- PostgreSQL database running locally
```

Comment:
To make it easy, let's add a comment describing how to run Postgres on docker desktop, with setup aligned to the settings in `debug-config.yaml`. Something like this:

```
docker run --name dev-postgres \
  -e POSTGRES_USER=radius_user \
  -e POSTGRES_PASSWORD=radius_pass \
  -e POSTGRES_DB=radius \
  -p 5432:5432 \
  -d postgres
```